src/resource/solidDataset.ts

```diff
@@ -444,7 +445,8 @@ function isUpdate(
     hasChangelog(solidDataset) &&
     hasResourceInfo(solidDataset) &&
     typeof solidDataset.internal_resourceInfo.sourceIri === "string" &&
-    solidDataset.internal_resourceInfo.sourceIri === url
+    normalizeServerSideIri(solidDataset.internal_resourceInfo.sourceIri) ===
+      normalizeServerSideIri(url)
```

Comment:
An alternative here, perhaps more readable, would be to provide an explicit comparison function that understands this particular context, e.g.:
```suggestion
    solidDatasetSourceEquals(solidDataset, url)
```

```javascript
function solidDatasetSourceEquals(solidDataset: SolidDataset, target: Iri): boolean {
  return solidDataset.internal_resourceInfo.sourceIri === target;
}
```

Reply:
709081950bf8327b2904c5101ddb687db76f58ea